src/Surfnet/StepupRa/RaBundle/Service/VettingService.php

```diff
@@ -195,8 +202,14 @@ public function sendSmsChallenge($procedureId, SendSmsChallengeCommand $command)
             $procedure->getSecondFactor()->secondFactorIdentifier
         );
 
+        $identity = $this->identityService->findById($procedure->getSecondFactor()->identityId);
+
+        if (!$identity) {
+            throw new DomainException("Second factor is coupled to an identity that doesn't exist");
```

Comment:
TL;DR: RuntimeException.

Small thing: imo, domain exceptions are really for when something exceptional occurs in the domain model (ie. business rules are directly violated). When this occurs, it is more likely the Middleware's data store is somehow in an inconsistent state and this would be a RuntimeException.

Other than that, if you tested it: :+1:.